.github/workflows/go-build-release.yaml

```diff
@@ -0,0 +1,77 @@
+name: Go Release Artifact Publisher
+
+on:
+  workflow_call:
+    inputs:
+      release_version:
+        required: true
+        type: string
+      repository_url:
+        required: true
+        type: string
+      branch:
+        required: true
+        type: string
+      service_name:
+        required: true
+        type: string
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    env:
+      GH_USER_NAME: ${{ github.actor }}
+
+    steps:
+      - name: Set ref
+        run: |
+            echo "REF=$([[$GITHUB_EVENT_NAME == 'release']] && echo refs/tags/${{ github.event.release.tag_name }} || echo ${{ inputs.branch }})" >> $GITHUB_ENV
+
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+         ref: ${{ env.REF }}
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.22.6
+
+      - name: Set environment variables
+        run: |
+          echo "APP_DIR=/go/src/github.com/org/repo" >> $GITHUB_ENV
```

Comment:
This is a constant value. Consider a way to describe this as an `env:` in the Build Artifacts step.